pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -286,6 +309,90 @@ func (m *ManifestWorkController) applyUnstructrued(data []byte, owner metav1.Own
 	return actual, true, err
 }
 
+func (m *ManifestWorkController) applyOwner(
+	gvr schema.GroupVersionResource,
+	namespace, name string,
+	deleteOption *workapiv1.DeleteOption,
+	owner metav1.OwnerReference,
+	recorder events.Recorder) error {
+	existing, err := m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(namespace).
+		Get(context.TODO(), name, metav1.GetOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	owners, _ := manageOwnerRef(gvr, namespace, name, deleteOption, existing.GetOwnerReferences(), owner)
+
+	if equality.Semantic.DeepEqual(owners, existing.GetOwnerReferences()) {
+		return nil
+	}
+
+	existing.SetOwnerReferences(owners)
+	_, err = m.spokeDynamicClient.
```

Comment:
I would only expect the secondary update for a removal flow if the returned value for the apply indicates a spurious ownerref.